Designed and Implemented Forum Screen

AbhiKul10 opened the pull request:
**Describe the Feature you Added**
Designed and Implemented Forum Screen

**How will your Feature Help?**
- There will be a separate channel for a particular topic.
- Users can enter channels and discuss anything related to the topic of the channel.
- only Admin can create new channels.
- Admin can assign any other user with admin rights using their email id.

Fixes #20 

<p align="center"  float="left">
<img src="https://user-images.githubusercontent.com/40908684/126154767-9635ddfa-2b16-4c36-9564-5aa1bd48a12b.gif" width="500" >

<img src="https://user-images.githubusercontent.com/40908684/126154810-263b3b55-7a21-46da-b2df-70d8ef7fd8cf.gif" width="500" >

<img src="https://user-images.githubusercontent.com/40908684/126154855-0fa24e7a-1f34-4006-b704-15a4dd92087f.gif" width="500" >
</p>

shehand commented on src/components/ModalHeaderNavigationBar/modalHeaderNavigationBar.js:
```diff
@@ -0,0 +1,23 @@
+import {Text, View, TouchableOpacity} from "react-native";
+import styles from "./styles";
+import Ionicons from "react-native-vector-icons/FontAwesome";
+export default class modalHeaderNavigationBar extends Component {
```
Normally we add a new line after imports and before class

shehand commented on src/components/SignupForm/SignupForm.js:
```diff
@@ -39,22 +39,35 @@ class SignupForm extends Component {
+          adminRights: false,
         };
         const userRef = firebase.firestore().collection("users");
+        const userRef1 = firebase.firestore().collection("userInfo");
```
shall we rename the variable name something more convenient and meaningful?

AbhiKul10 replied:
done

shehand commented on src/components/SignupForm/SignupForm.js:
```diff
@@ -39,22 +39,35 @@ class SignupForm extends Component {
+            id: uid,
+          })
+          .then(() => {
+            console.log("Document successfully written!");
```
logging to console is not the best thing to do here.

shehand commented on src/config/styles.js:
```diff
@@ -7,3 +7,4 @@ export const COLOR_LIGHT_SECONDARY = "#79AEE0";
 // IMAGES
 export const LOGO = require("../../assets/Ask-SCORE-Logo.png");
 export const NO_IMAGE = require("../../assets/no-image.png");
+export const NO_IMAGE_LINK = "https://bit.ly/3il86S9";
```
better to have the image in the project itself as there is no guarantee on the URL

shehand commented on src/screens/Addroom/Addroom.js:
```diff
@@ -0,0 +1,232 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-shadow */
+/* eslint-disable react-native/no-inline-styles */
+/* eslint-disable no-alert */
```
let's remove these

shehand commented on src/screens/Addroom/Addroom.js:
```diff
@@ -0,0 +1,232 @@
+            messageTime: timeDate.format("lll"),
+            imageURL: profilePresent
+              ? doc.data().downloadURL
+              : "https://bit.ly/3il86S9",
```
why aren't you using the const variable here instead?

shehand commented on src/screens/ChatScreen/ChatScreen.js:
```diff
@@ -0,0 +1,327 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-alert */
+/* eslint-disable react-native/no-inline-styles */
+/* eslint-disable no-unused-vars */
```
no need

AbhiKul10 replied:
removed

shehand commented on src/screens/ChatScreen/ChatScreen.js:
```diff
@@ -0,0 +1,327 @@
+    };
+
+    const taskError = snapshot => {
+      alert("Error Occured");
```
shall we specify the error here instead of having a common alert?

shehand commented on src/screens/ChatScreen/ChatScreen.js:
```diff
@@ -0,0 +1,327 @@
+    } = this.state;
+    return (
+      <>
+        <SafeAreaView style={{flex: 0, backgroundColor: "#51AD28"}} />
```
It would be great of you can move all the css styles into a single styles file for this view.

shehand commented on src/screens/ForumScreen/ForumScreen.js:
```diff
@@ -1,14 +1,146 @@
+                    this.props.navigation.navigate("AddChat");
+                  }}>
+                  <Text
+                    style={{
```
Let's move all the styles into the separate styles file of this component

AbhiKul10 replied:
done

shehand reviewed (changes requested):
I'm not 100% sure whether we should do this, but do we need to implement the delete chat room functionality also?

AbhiKul10 commented:
> I'm not 100% sure whether we should do this, but do we need to implement the delete chat room functionality also? And I did not see any unit testing for this feature.

I have implemented it! Chat room deletion rights are only available for the one who has administrator rights. I have added the test for the component, other tests are failing due to the error we faced before, Here is the screenshot.

![Screenshot 2021-07-21 at 3 30 27 PM](https://user-images.githubusercontent.com/40908684/126471668-76b1382b-6535-4096-b283-25538d16ffcd.png)

I have tested this on CircleCI and it's working, once our project is added on CircleCI, I will push all the test cases that are failing due to the above error.

shehand commented:
> > I'm not 100% sure whether we should do this, but do we need to implement the delete chat room functionality also? And I did not see any unit testing for this feature.
> 
> I have implemented it! Chat room deletion rights are only available for the one who has administrator rights. I have added the test for the component, other tests are failing due to the error we faced before, Here is the screenshot.

Just update the PR with a gif with that feature also.
